### DISCUSSION TITLE
ignoreDeps is not considered as a valid package rule selector

### DISCUSSION THREAD

**Discussion by moltar** [renovatebot/renovate#25107] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.82.0

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

I have package rules setup per package for each package in the monorepo to ignore some of the deps:

```json
    {
      "ignoreDeps": [
        "@types/node",
        "@typescript-eslint/eslint-plugin",
        "@typescript-eslint/parser",
        "eslint"
      ],
      "groupName": "foo: ignore pinned",
      "includePaths": [
        "packages/foo/package.json"
      ]
    },
```

However, config validation fails with:

> Each packageRule must contain at least one match* or exclude* selector.

But the docs for `excludePackageNames` state:

> Important: Do not mix this up with the option ignoreDeps. Use ignoreDeps instead if all you want to do is have a list of package names for Renovate to ignore.

---

Does `ignoreDeps` need to be in the selector list in `lib/config/validation.ts`?

### Relevant debug logs

<details><summary>Logs</summary>

```
Important: Do not mix this up with the option ignoreDeps. Use ignoreDeps instead if all you want to do is have a list of package names for Renovate to ignore.
```

</details>

### Have you created a minimal reproduction repository?

Placeholder value, please select the correct response from the dropdown

**Comment by rarkins** [comment, marked as answer]
You seem to be mixing up ignoreDeps with excludePackageNames. ignoreDeps is not valid within packageRules

**Reply by moltar** [reply]
Thanks, @rarkins you are totally right.
